### PR TITLE
Support the `--version` flag

### PR DESCRIPTION
**Pull request by bamarni**
It displays the CLI as well as installed plugins versions.

There are also version information regarding the currently attached DC/OS cluster, if any.

https://jira.mesosphere.com/browse/DCOS_OSS-3702

### CODE REVIEW

**Comment by ArmandGrillet** [comment]
LGTM but:
```
cli (DCOS_OSS-3702) $ CC="" go build -o ./dcos-go ./cmd/dcos
cli (DCOS_OSS-3702) $ ./dcos-go
Usage:
  dcos [command]

Available Commands:
  auth
  backup      Manage DC/OS Backups
  cluster
  config
  help        Help about any command
  kafka       Kafka DC/OS CLI Module
  security    DC/OS security related commands

Flags:
  -v, -- count   verbosity (-v or -vv)
  -h, --help     help for dcos

Use "dcos [command] --help" for more information about a command.
cli (DCOS_OSS-3702) $ ./dcos-go --version
dcoscli.version=SNAPSHOT
dcoscli.dcos-enterprise-cli.version=1.3.0
dcoscli.kafka.version=2.3.0-1.0.0
dcos.version=1.10.7
dcos.commit=1f1403295d737350e850ec20c00425fd9aed21ad
dcos.bootstrap-id=9bb7ff0633725bbadd0707e4516c495162f65067
```
Shouldn't it display a new flag when doing `./dcos-go`?

File: pkg/plugin/manager.go [hunk 1]
```diff
@@ -130,7 +131,14 @@ func (m *Manager) loadPluginFromPackage(plugin *Plugin) error {
 	}
 
 	// We transfer the information we have from the package to the plugin.
-	plugin.Name = pkg.Name
+	// Packages called `dcos-*-cli` get a simpler name as it becomes redundant
+	// in the plugin context. Eg. `dcos-enterprise-cli` -> `enterprise`.
+	matches := regexp.MustCompile("^dcos-(.+)-cli$").FindStringSubmatch(pkg.Name)
```
Comment (br-lewis) [inline pkg/plugin/manager.go]: While it is redundant, it's also the name of the package that was installed so I would support keeping it. Are there any other known subcommands that do that?

**Comment by bamarni** [comment]
@ArmandGrillet : good point. It is indeed missing as outside of cobra. We will most likely customize the whole help message to follow our guidelines so I'd do this in its own ticket, the same way we did for #1253. I've created https://jira.mesosphere.com/browse/DCOS_OSS-3705

**Comment by br-lewis** [comment]
You could add a dummy version flag to the root command to have it show up in the help message.

**Comment by bamarni** [comment]
@br-lewis : see https://jira.mesosphere.com/browse/DCOS_OSS-3705, the help message will be worked on separately. It will probably not rely on flags introspection.

**Pull request merged**